### DISCUSSION TITLE
requiv not within limits

### DISCUSSION THREAD

**Discussion by jvillasr** [phoebe-project/phoebe2#765] [Questions]
Hi, I'm trying to fit the light curve of a binary. I have the orbital solution, I'm also giving the inclination, radii and temperatures, while fitting for requiv@secondary and teff@secondary.

For some inclinations I get this warning:

> WARNING triangles on ['primary'] may be larger than the entire bodies of ['secondary'], resulting in inaccurate eclipse detection.  Check values for requiv of ['secondary'] and/or ntriangles of ['primary'].  If your system is known to NOT eclipse, you can set eclipse_method to 'only_horizon' to circumvent this check.  If not addressed, this warning will continue to be raised and will throw an error at run_compute.

which I solved by setting `b['eclipse_method'] = 'only_horizon'` (system should not be eclipsing)

but then I get this other warning:

> WARNING received error while setting values: value of requiv=-0.19449999999999257 solRad not within limits of [<Quantity 1.e-06 solRad>, None]. lnprobability=-inf

I don't know why Phoebe is trying to use negative radii, and it gives a very low value after computation:

> ParameterSet: 2 parameters
>    teff@secondary@star@component: 32010.716611730422 K
>    requiv@secondary@star@compo...: 1.2151875975210301e-06 solRad

Temperature is OK but radius should be around 5-7 solRad. In other models I also got values of requiv@secondary=0.48 and teff@secondary=411. So something weird is going on. And when I compute a model, previous to the solver, the solution seems reasonable. Any suggestions?

**Comment by kecnry** [comment]
> For some inclinations I get this warning:

>> WARNING triangles on ['primary'] may be larger than the entire bodies of ['secondary'], resulting in inaccurate eclipse detection. Check values for requiv of ['secondary'] and/or ntriangles of ['primary']. If your system is known to NOT eclipse, you can set eclipse_method to 'only_horizon' to circumvent this check. If not addressed, this warning will continue to be raised and will throw an error at run_compute.

> which I solved by setting b['eclipse_method'] = 'only_horizon' (system should not be eclipsing)

Just be careful that the solver does not drive the solution to a case where it would result in eclipses - since the synthetic light curve will now never show eclipses and so won't necessarily be penalized when comparing against the observations.

> but then I get this other warning:

>> WARNING received error while setting values: value of requiv=-0.19449999999999257 solRad not within limits of [<Quantity 1.e-06 solRad>, None]. lnprobability=-inf

> I don't know why Phoebe is trying to use negative radii

This warning alone isn't a concern.  The solver (optimizer or sampler) can test any value for the parameters, but will immediately reject that step with `lnprobability=-inf` if those are outside parameter limits, before even attempting to run the forward model.

> and it gives a very low value after computation

This though is a problem.  You can always impose a lower-limit on the radius via a non-informative uniform prior, but I'm guessing that alone will only result in the solver "running up" against whatever artificial limit you set (you can see that is essentially already happening with phoebe's limit of 1e-6... it seems that it "wants" to go to a radius of zero).  My suggestion would be to look at the residuals _before_ running the solver to try to understand why the optimizer/sampler is driving the radius to small values.  For example: Is it doing that to try to change the overall light contribution between the two stars and finds it easier with the radius than temperature?  Is it just trying to decrease overall flux values?  If either of these are the case - double check how you're handling [passband luminosity](http://phoebe-project.org/docs/latest/tutorials/pblum.ipynb) and [third light](http://phoebe-project.org/docs/latest/tutorials/l3.ipynb) and consider letting the temperature/radius of the other star vary as well.  You can then try to introduce more flexibility into the model with relevant parameters to give the solver freedom to account for those residuals without decreasing the radius so much.

I would also double-check the masses to make sure they are reasonable for your system as the combination of mass and radius will drive loggs, which in turn affect the lookups from atmosphere tables.

Hopefully this helps!

**Reply by jvillasr** [reply]
Hi Kyle, thanks for your reply, that's very helpful.

Unfortunately, I couldn't detect what was causing the small radius values. Computing a model before the optimiser didn't offer any clues, at least to my rookie eye, both fit and residuals seemed reasonable given the noisy light curve. Allowing for more flexibility did solve the issue though, so I tried different combinations of fixed and fitted parameters, and for all of them I got values in the expected range for the masses, temperatures, and radii, so I'm satisfied with the result!

For the passband luminosity I'm using `b.set_value(qualifier='pblum_mode', context='dataset', dataset='lc01', value='dataset-scaled')`. And for third light, the default value.
Other parameters I'm using:
```
b.set_value_all('atm', value='blackbody')
b.set_value_all('gravb_bol', value=1)
b.set_value_all('irrad_frac_refl_bol', value=1)
b.set_value_all('ld_mode_bol', value='manual')
b.set_value_all('ld_mode', value='manual')
```
Essentially, the values suggested by Phoebe for massive stars. 
Would you recommend to look at any of these parameters into more detail?

**Reply by kecnry** [reply]
ok, I'm glad to hear adding more parameters seems to have done the trick!
